beard/clustering/blocking.py

```diff
@@ -84,41 +97,54 @@ def _validate(self, X, blocks):
 
         return X, blocks
 
-    def _fit(self, X, blocks):
+    def _fit(self, X, y, blocks):
         """Fit base clustering estimators on X."""
         self.labels_ = -np.ones(len(X), dtype=np.int)
         offset = 0
 
         for b in np.unique(blocks):
-            # Fit on the block
+            # Select data from block
             mask = (blocks == b)
             X_mask = X[mask, :]
 
+            if y is not None:
+                y_mask = y[mask]
+            else:
+                y_mask = None
+
             if self.affinity == "precomputed":
                 X_mask = X_mask[:, mask]
 
-            if self.fit_:
-                cluster = clone(self.base_estimator)
-                cluster.fit(X_mask)
-
+            # Fit a clusterer on the selected data
+            if len(X_mask) == 1:
+                clusterer = _SingleClustering()
+            elif self.fit_:
+                clusterer = clone(self.base_estimator)
             elif self.partial_fit_:
                 if b in self.clusterers_:
```

Comment:
If we are iteratin over the result `np.unique`, can `b in self.clusteres_` be true?